src/js/common/schemaform/helpers.js

```diff
@@ -420,7 +427,11 @@ export function setHiddenFields(schema, uiSchema, formData) {
   }
 
   if (updatedSchema.type === 'array') {
-    const newSchema = setHiddenFields(updatedSchema.items, uiSchema.items, formData);
+    // Our current approach is to sometimes modify the schema based on form data
+    // This breaks down with arrays, because all items share the same schema
+    // So, this is technically not correct, but the last row is probably what's visible
+    // for a user, so it should be the least bad for now
+    const newSchema = setHiddenFields(updatedSchema.items, uiSchema.items, formData, path.concat(formData.length - 1));
```

Comment:
So based on your comment, this will only set hidden fields for the last item in an array? An arrays are present for things like schools or employment periods where they could one more than one?